roles/provision_vcenter/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Validate Inputs
+  ansible.builtin.include_tasks: validate_inputs.yml
+
+- name: Check If VCenter Already Exists
+  community.vmware.vmware_guest_info:
+    hostname: "{{ provision_vcenter_auth_hostname }}"
+    username: "{{ provision_vcenter_auth_username }}"
+    password: "{{ provision_vcenter_auth_password }}"
+    port: "{{ provision_vcenter_auth_port | default(omit) }}"
+    validate_certs: "{{ provision_vcenter_auth_validate_certs | default(omit) }}"
+    datacenter: "{{ provision_vcenter_vc_target_datacenter | default('') }}"
+    name: "{{ provision_vcenter_vm_name }}"
+  register: _vcenter_lookup
+  failed_when: false
+
+- name: Create Instance From ISO
+  ansible.builtin.include_tasks: deploy_from_iso.yml
+  when: _vcenter_lookup.instance is not defined
```

Comment:
You always run the deployment of the Vcenter. I am not sure user would expect this as default behaviour. What about adding `redeploy` parameter, which will redeploy the vcenter instance in case for example we can see VM exists and the URL is responding? And we don't redeploy the instance by default.

Reply:
> `when: _vcenter_lookup.instance is not defined` this condition should mean that we only deploy if the instance/virtual machine specified by the user does not exist. So we dont deploy everytime, just if the vcenter instance does not exist.

I think in case VM exists the `vmware_guest` module still return `instance` parameter, see:
https://github.com/ansible-collections/community.vmware/blob/main/plugins/modules/vmware_guest.py#L3294

> 
> I may be misunderstanding, but it sounds like you think the default action should be to not deploy, even if the instance is missing?

No, I meant to make it work the same as you've intended, but I think checking the `instance` may not work.

Reply:
ah ok i see. ill refine the check

Reply:
I did some testing and the `instance` attribute is not returned when the VM does not exist. However, there are other instances when `instance` doesn't exist and we actually want to alert the user of an error. So i added some additional error handling around this. I also made the conditional for deploying vcenter handle if `instance` is returned but empty, since maybe there is another case where that occurs